packages/cli-v3/src/dev/lock.ts

```diff
@@ -0,0 +1,87 @@
+import path from "node:path";
+import { readFile } from "../utilities/fileSystem.js";
+import { tryCatch } from "@trigger.dev/core/utils";
+import { logger } from "../utilities/logger.js";
+import { writeFile } from "node:fs/promises";
+import { existsSync, unlinkSync } from "node:fs";
+import { onExit } from "signal-exit";
+
+const LOCK_FILE_NAME = "dev.lock";
+
+export async function createLockFile(cwd: string) {
+  const currentPid = process.pid;
+  const lockFilePath = path.join(cwd, ".trigger", LOCK_FILE_NAME);
```

Comment:
We should check the `.trigger` folder exists first, and create it if it doesn't.